In [1]:
#importing necessary libraries
import pandas as pd
import numpy as np
from sklearn import preprocessing
from sklearn import tree
from sklearn.model_selection import train_test_split
from sklearn.naive_bayes import *
from sklearn.metrics import accuracy_score
from sklearn.metrics import confusion_matrix

#loading working file in variable data
data=pd.read_csv("C:/Users/Dell/Downloads/train.csv")

In [18]:
label_encoder=preprocessing.LabelEncoder()
#Converting Object type variables to numerical type,cleaning the data
encoded_sex=label_encoder.fit_transform(data["Sex"])
data["Sex"]=encoded_sex

encoded_Embarked=label_encoder.fit_transform(data["Embarked"])
data["Embarked"]=encoded_Embarked
data1=data.drop(["Name","Cabin","Ticket","PassengerId","Age","Fare"],axis=1)

In [19]:
data1.head()

Survived  Pclass  Sex  SibSp  Parch  Embarked
0         0       3    1      1      0         2
1         1       1    0      1      0         0
2         1       3    0      0      0         2
3         1       1    0      1      0         2
4         0       3    1      0      0         2

In [22]:
for column in data1:
    t=data1.drop([column],axis=1)
    y=data1[column]
    X=t
    #Declaring Naive bayes function
    X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.3,random_state=0)
    
    clf=BernoulliNB()
    y_pred=clf.fit(X_train,y_train).predict(X_test)
    
    #Finding Accuracy of model
    print(X)
    print(y)
    print(accuracy_score(y_test,y_pred,normalize=True))
    #Building Confusion matrix of model
    print(confusion_matrix(y_test,y_pred))

     Pclass  Sex  SibSp  Parch  Embarked
0         3    1      1      0         2
1         1    0      1      0         0
2         3    0      0      0         2
3         1    0      1      0         2
4         3    1      0      0         2
..      ...  ...    ...    ...       ...
884       2    1      0      0         2
885       1    0      0      0         2
886       3    0      1      2         2
887       1    1      0      0         0
888       3    1      0      0         1

[889 rows x 5 columns]
0      0
1      1
2      1
3      1
4      0
      ..
884    0
885    1
886    0
887    1
888    0
Name: Survived, Length: 889, dtype: int64
0.7715355805243446
[[131  26]
 [ 35  75]]
     Survived  Sex  SibSp  Parch  Embarked
0           0    1      1      0         2
1           1    0      1      0         0
2           1    0      0      0         2
3           1    0      1      0         2
4           0    1      0      0         2
..        ...  ...    ...    ...       ...


__Hence , the accuracy score with Survived as dependent variable and all other are independent variables is most i.e. 77%